source/Plugins.cpp

```diff
@@ -0,0 +1,82 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	map<string, bool> settings;
+
+	void LoadSettings(const string &path)
+	{
+		DataFile prefs(path);
+		for(const DataNode &node : prefs)
+			settings[node.Token(0)] = (node.Size() == 1 || node.Value(1));
+	}
+}
+
+
+
+void Plugins::Load()
+{
+	// Global plugin settings
+	LoadSettings(Files::Resources() + "plugins.txt");
```

Comment:
Any reason for not using `preferences.txt` to store the list of enable plugins? It seems fitting.

Reply:
No particular reason.  I would need to extend preferences to differentiate setting or plugin.  Having a plugins.txt felt cleaner because I could define default plugins settings for global plugins.  It can have its own validation separate from preferences.  plugins.txt felt like a natural item next to the plugins folder